src/sources/stdin.rs

```diff
@@ -40,32 +36,58 @@ impl SourceConfig for StdinConfig {
         _globals: &GlobalOptions,
         out: mpsc::Sender<Event>,
     ) -> crate::Result<super::Source> {
-        Ok(stdin_source(stdin(), self.clone(), out))
+        Ok(stdin_source(
+            io::BufReader::new(io::stdin()),
+            self.clone(),
+            out,
+        ))
     }
 
     fn output_type(&self) -> DataType {
         DataType::Log
     }
 }
 
-pub fn stdin_source<S>(stream: S, config: StdinConfig, out: mpsc::Sender<Event>) -> super::Source
+pub fn stdin_source<R>(mut stdin: R, config: StdinConfig, out: mpsc::Sender<Event>) -> super::Source
 where
-    S: AsyncRead + Send + 'static,
+    R: Send + io::BufRead + 'static,
 {
     Box::new(future::lazy(move || {
         info!("Capturing STDIN");
 
         let host_key = config.host_key.clone().unwrap_or(event::HOST.to_string());
         let hostname = hostname::get_hostname();
-
-        let source = FramedRead::new(
-            stream,
-            BytesDelimitedCodec::new_with_max_length(b'\n', config.max_length),
-        )
-        .map(move |line| create_event(line, &host_key, &hostname))
-        .map_err(|e| error!("error reading line: {:?}", e))
-        .forward(out.sink_map_err(|e| error!("Error sending in sink {}", e)))
-        .map(|_| info!("finished sending"));
+        let (mut tx, rx) = futures::sync::mpsc::channel(1024);
+
+        thread::spawn(move || {
+            println!("Loaded thread");
+            loop {
+                let mut buf = String::new();
+                match stdin.read_line(&mut buf) {
+                    Err(e) => {
+                        error!("Error reading from source: stdin: {}", e);
```

Comment:
Since we are using the `tracing` crate for logging this is the preferred way to setup printing errors.
```suggestion
                        error!(message = "Unable to read from source.", error = %e);
```